tasks/system_probe.py

```diff
@@ -618,6 +618,13 @@ def build(
     """
     Build the system-probe
     """
+    if python_runtimes:
```

Comment:
One suggestion, wouldn't it be better to extract these calls to a single `warn_deprecation_if_used(python_runtimes)` common function to be used everywhere? It'd also make it easier to track later which parameters are still active)

Reply:
very good point! Done locally, I'll push after addressing some of the other remarks